File: CSharp/1901.cs
using System;
using System.Linq;
using System.Collections.Generic;

class URI
{
    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());
        int[][] floresta = new int[n][];
        int[] celula;
        HashSet<int> listao = new HashSet<int>();

        for(int i=0;i<n;i++)
        {
            floresta[i] = Console.ReadLine().Split().Select(int.Parse).ToArray();
        }
        for(int i=0;i<n*2;i++)
        {
            celula = Console.ReadLine().Split().Select(int.Parse).ToArray();
            listao.Add(floresta[celula[0]-1][celula[1]-1]);
        }
        Console.WriteLine(listao.Count());
    }
}

File: Python/1901.py
n=int(input())
listao =set()
floresta=[]
for i in range(n):
    floresta.append(list(map(int,input().split())))
for i in range(2*n):
    celula =  list(map(int,input().split()))
    listao.add(floresta[celula[0]-1][celula[1]-1])
print(len(listao))